src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -4079,6 +4079,13 @@ def aks_agentpool_upgrade(cmd, client, resource_group_name, cluster_name,
             'node version please use the "--node-image-only" option only.'
         )
 
+    if max_surge and node_image_only:
```

Comment:
comment that upgrade can't accept agentpool put fields like max surge.

Reply:
Added comment:
`Note: we exclude this option because node image upgrade can't accept nodepools put fields like max surge`